src/boot/efi/stub.c

```diff
@@ -912,16 +976,18 @@ static EFI_STATUS run(EFI_HANDLE image) {
         generate_embedded_initrds(loaded_image, sections, initrds);
         lookup_embedded_initrds(loaded_image, sections, initrds);
 
+        /* Measures ucode addons and populates all_initrds with the final order of all initrds */
+        measure_ucode_addons_and_create_final_list(initrds, &all_initrds, &n_all_initrds, ucode_addons, n_ucode_addons, &sections_measured);
+
         /* Export variables indicating what we measured */
         export_pcr_variables(sections_measured, parameters_measured, sysext_measured, confext_measured);
 
         /* Combine the initrds into one */
         _cleanup_pages_ Pages initrd_pages = {};
         struct iovec final_initrd;
-        if (initrds_need_combine(initrds)) {
-                /* If we have generated initrds dynamically or there is a microcode initrd, combine them with
-                 * the built-in initrd. */
-                err = combine_initrds(initrds, _INITRD_MAX, &initrd_pages, &final_initrd.iov_len);
+        if (n_all_initrds > 1) {
```

Comment:
Is this actually true with https://github.com/uapi-group/specifications/pull/83?

Reply:
Hmm good point. Now that I think about it, was the stub already broken for UKIs without any initrds?

Before this PR, we would pass an empty iovec, and with this PR we pass NULL to `linux_exec` - both of them will fail the same on `assert(iovec_is_valid(initrd));` in `linux_exec`.

I am happy to pass something else if that is useful, but I don't want to also put fixing `linux_exec` in scope here, as is is kinda unrelated. Is that fair?

Reply:
That's fair, let's keep it as is then.